### PR TITLE
Add activate/deactivate support for packed environment variables

### PR DESCRIPTION
**Pull request by Jrice1317**
### Description
Addresses [Issue #397](https://github.com/conda/conda-pack/issues/397)

This PR adds support for preserving environment variables configured through `environment.yaml` when creating packed environments.

Previously, environment variables stored in `conda-meta/state` were not fully restored when activating a packed environment. This change ensures those variables are available after extraction and properly cleaned up on deactivation.

### Changes

- Include `conda-meta/state` in packed environments
- Add support for setting environment variables during activation
- Add matching deactivation behavior to unset/restore environment variables
- Support `posix`, `windows`, and shell flavors

## Example 1

Given an environment.yml configured with:

```
variables:
  MY_VAR: hello
```
  After packing and extracting the env,
  ```
  echo $MY_VAR      # prints hello
  ```
  and the variable will be restored after deactivating the environment.
 

## Example 2

Given a pre-existing variable:
```
MY_VAR=hello
```
An environment.yml configured with: 
```
variables:
  MY_VAR: goodbye
```
  After packing and extracting the env,
  ```
  echo $MY_VAR      # updates to goodbye
  ```
  and the variable will be restored to its original state after deactivating the environment.
```
echo $MY_VAR        # prints hello
  ```

### CODE REVIEW

File: .github/workflows/main.yml
Comment (xhochy) [inline .github/workflows/main.yml:1]: Revert the changes in this file please.

File: conda_pack/core.py [hunk 1]
```diff
@@ -1290,6 +1364,36 @@ def finish(self):
...
+                        _SH_DEACTIVATE_TEMPLATE,
+                        "sh"
+                    )
+                    self._write_env_var_scripts(
+                        env_vars,
+                        lambda v: v.replace("\\", "\\\\").replace("'", "\\'"),
+                        _FISH_ACTIVATE_TEMPLATE,
+                        _FISH_DEACTIVATE_TEMPLATE,
+                        "fish"
+                    )
```
Comment (ryanskeith) [inline conda_pack/core.py:1395]: This could should be indented.  Right now the code writes out the environment var scripts always.  The above code block will only write out scripts if the two conditions are not met and these should follow the same.  Adding `conda-meta/state` should help with the other case though it may have already been tracked.

File: conda_pack/scripts/posix/activate.fish [hunk 2]
```diff
@@ -43,6 +43,15 @@ function _activate
 end
 
 function deactivate -d 'Exit conda mode and return to the normal environment.'
+    set -l _script_dir "$CONDA_PREFIX/etc/conda/deactivate.d"
```
Comment (ryanskeith) [inline conda_pack/scripts/posix/activate.fish:46]: Does the code need to protect if CONDA_PREFIX exists like the other deactivate script?

File: conda-recipe/meta.yaml [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 package:
   name: conda-pack
-  version: {{ version_base }}.{{ GIT_DESCRIBE_NUMBER }}+{{ GIT_DESCRIBE_HASH }}
+  version: {{ version_base }}
```
Comment (ryanskeith) [inline conda-recipe/meta.yaml:5]: Why the change?

File: testing/env_yamls/env_vars.yml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+channels:
+  - conda-forge
+
+dependencies:
+  - requests
+
+variables:
+  MY_EXISTING_VAR: hello
+  MY_SPECIAL_VAR: "red=|<>!&^'%123"
+  MY_QUOTED_VAR: 'say "hello"'
```
Comment (ryanskeith) [inline testing/env_yamls/env_vars.yml:12]: What about the case that has curly braces?

```
MY_CURLY_VAR: '${some_var}'
```

**Review by ryanskeith — COMMENTED** [review]
Looks like a good direction.  I am concerned on a couple things that I noted.  First and foremost is the conda pack mode where it intends that target is a conda environment and the scripts aren't meant to be written.  Currently the env scripts will be written which might interfere with the conda-meta/state file.  The current solution will work great in the absence of a real conda central command line. 

I need to think about it more but I am wondering if there is a simpler way that would make it easier on the user to edit variables.

I had a few more questions and comments I feel need to be addressed before moving forward.

File: conda_pack/core.py (hunk 1 above)
Comment (Jrice1317) [reply conda_pack/core.py]: Thanks for bringing this up! It took me back to Set Theory for a bit. 😅  At first, I was thinking the scripts wouldn't be written if `has_dest` was true but `has_conda` was false, but after thinking about it a little harder and using the help of my friends Google and Kilo, the only time it skips writing is when both `has_dest` and `has conda` are true. So, indenting makes sense here.

File: conda_pack/scripts/posix/activate.fish (hunk 2 above)
Comment (Jrice1317) [reply conda_pack/scripts/posix/activate.fish]: It probably should, so I've added the guard. Thanks!

File: conda-recipe/meta.yaml (hunk 3 above)
Comment (Jrice1317) [reply conda-recipe/meta.yaml]: Whoops, that was an accident! It's leftover from testing. Reverting.

File: testing/env_yamls/env_vars.yml (hunk 4 above)
Comment (Jrice1317) [reply testing/env_yamls/env_vars.yml:12]: That's a great edge case! Could you please clarify how it should be handled? Right now curly braces are treated literally. Should we allow expansion or is literal the intended behavior and I should add it to the tests to make sure that's the case? I wasn't sure if we needed to worry about injections here.

Comment (ryanskeith) [reply testing/env_yamls/env_vars.yml:12]: I think it should be added as a test case.  I am not worried about injection as it is the users setting it up but I could see it be quite handy for conda-pack to allow it.  It should be fine, I hope. I just wanted to identify a common case.

**Review by ryanskeith — APPROVED** [review]
Thank you so much @Jrice1317
